verteiler/src/main/java/io/datanerds/verteiler/BlockingQueueConsumer.java

```diff
@@ -120,21 +119,20 @@ private void createProcessor(TopicPartition partition) {
     }
 
     private Consumer<K, V> createKafkaConsumer() {
-        addClientIdIfNotPresent(kafkaConfig);
-        addOffsetResetConfigIfNotPresent(kafkaConfig);
+        setPropertyDefaultIfNotPresent(kafkaConfig, CLIENT_ID_CONFIG, this::getClientId);
+        setPropertyDefaultIfNotPresent(kafkaConfig, AUTO_OFFSET_RESET_CONFIG, () -> "earliest");
         return new KafkaConsumer<>(kafkaConfig);
     }
 
-    private void addClientIdIfNotPresent(Properties props) {
-        if (!props.contains(CLIENT_ID_CONFIG)) {
-            props.put(CLIENT_ID_CONFIG, getClientId());
-        }
-    }
-
-    private void addOffsetResetConfigIfNotPresent(Properties props) {
-        if (!props.contains(AUTO_OFFSET_RESET_CONFIG)) {
-            props.put(AUTO_OFFSET_RESET_CONFIG, "earliest");
+    private void setPropertyDefaultIfNotPresent(
+            Properties props,
+            String propertyName,
+            Supplier<String> defaultValue
+    ) {
+        if (props.contains(propertyName)) {
```

Comment:
My original motivation was having the [GuardClause](http://wiki.c2.com/?GuardClause) instead of the negated boolean condition, which I find rather irritating to read.